camera/set-up-vm.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+
+print_usage_and_die() {
+    echo >&2 "Usage: $0 [<vm-files-directory>]"
+    exit 1
+}
+
+# Force running from the same directory
+if [ "$PWD" != "$(dirname "$(realpath "$0")")" ]; then
+    echo >&2 "Error: Script called from a different directory" 
+    print_usage_and_die
+fi
+
+# Check for custom VM_DIR
+if [ "$#" -eq 1 ]; then
+    VM_DIR="$1"
+else
+    VM_DIR="vm-files"
+fi
+
+PROJECT_NAME=ivory-beam
+RASPBIAN_ISO_URL="https://downloads.raspberrypi.org/rpd_x86/images/rpd_x86-2020-02-14/2020-02-12-rpd-x86-buster.iso"
+RASPBIAN_BUILD="2020-02-12-rpd-x86-buster"
+
+VM_NAME="${PROJECT_NAME}-camera"
+VM_ISO_PATH="${VM_DIR}/${RASPBIAN_BUILD}.iso"
+VM_DISK_FILE_PATH="${VM_DIR}/${VM_NAME}-disk.vdi"
+
+mkdir -p "${VM_DIR}"
+echo "Using vm dir: ${VM_DIR}"
+
+echo "Downloading image..."
+wget -q --show-progress "${RASPBIAN_ISO_URL}" -O "${VM_ISO_PATH}"
+
+echo "Creating vm..."
+vboxmanage createvm --name "${VM_NAME}" --ostype "Linux" --register --basefolder "${CURRENT_DIR}" && # Linux OS type is "Other Linux (32-bit)"
+vboxmanage modifyvm "${VM_NAME}" --memory 6144 --vram 128 && # 6 GB RAM, 128 MB VRAM
```

Comment:
We cannot use th VM with so much RAM as it is not practical to take so much resources from host computer + raspberry has 2 GBs of RAM so I don't see any reason to give so much memory to VM (if we want fast compilation/better performance on tests we can test it on local machines with opencv)

Reply:
I missed differences in host systems. c07f7e6 comes with extra flags for RAM and disk size. Also, RAM default is now 512 MB to match Raspberry Pi 3 A+ ([official spec](https://www.raspberrypi.org/products/raspberry-pi-3-model-a-plus/)).

Scripts are needed nevertheless, you can never guarantee your environment will exactly match one on the device. Trivial example: if somebody uses non-debian derivative they can't run `apt-get` commands for updating Pi libraries, not speaking of version mismatches and so on...